diff --git a/lib/Padre/Wx/Constant.pm b/lib/Padre/Wx/Constant.pm
index 99e55a2e7..8c5b1a50b 100644
--- a/lib/Padre/Wx/Constant.pm
+++ b/lib/Padre/Wx/Constant.pm
@@ -270,7 +270,6 @@ use constant WANT => qw{
 	:listctrl
 	:locale
 	:menu
-	:miniframe
 	:misc
 	:notebook
 	:ownerdrawncombobox
diff --git a/lib/Padre/Wx/Main.pm b/lib/Padre/Wx/Main.pm
index bec984e2f..fecfe681f 100644
--- a/lib/Padre/Wx/Main.pm
+++ b/lib/Padre/Wx/Main.pm
@@ -60,6 +60,8 @@ use Padre::Wx::Role::Timer    ();
 use Padre::Wx::Role::Idle     ();
 use Padre::Locale::T;
 use Padre::Logger;
+use File::Which 'which';
+
 our $VERSION    = '1.02';
 our $COMPATIBLE = '0.91';
 our @ISA        = qw{
@@ -2955,11 +2957,14 @@ sub run_command {
 		} elsif (Padre::Constant::UNIX) {
 
 			if ( defined $ENV{COLORTERM} ) {
-				if ( $ENV{COLORTERM} eq 'gnome-terminal' ) {
-
+   				my @terminals = ();
+       				push @terminals, which('gnome-terminal');
+			 	push @terminals, which('x-terminal-emulator');
+     				push @terminals, which('konsole');
+				if ( @terminals ) {
 					#Gnome-Terminal line format:
 					#gnome-terminal -e "bash -c \"prove -lv t/96_edit_patch.t; exec bash\""
-					system qq($ENV{COLORTERM} -e "bash -c \\\"$cmd; exec bash\\\"" & );
+					system $terminals[0], qq(-e "bash -c \\\"$cmd; exec bash\\\"" & );
 				} else {
 					system qq(xterm -sb -e "$cmd; sleep 1000" &);
 				}
